pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -487,6 +513,16 @@ func estimateGracefulTerminationForPods(kubeClient clientset.Interface, ns strin
 	return estimate, nil
 }
 
+func unstructuredToPod(obj *runtime.Unstructured) (*api.Pod, error) {
```

Comment:
@caesarxuchao I want to highlight this specially. Before this PR we were using kubeclient to list pods. With this change I am only passing NamespaceInterface here and using the dynamic client to list pods. I cant pass CoreInterface since CoreInterface from federation clientset is different than the CoreInterface in kube clientset (for ex: federation does not have component statuses). So I had 2 options:
* Pass both NamespaceInterface and PodInterface
* Use the dynamic client to list pods.

The second option I think is better. It allows better extensibility when more resources support graceful termination.